Optionally log objects instead of messages

bllanos opened the pull request:
Requested reviewers: @calvinwiebe 

### Changes

- Added a new `logObject` option to the configuration function.
- When true, `time-me` will output a `{prefix, elapsed}` object instead of a string version of the same information.

### Testing Performed
- Updated unit tests to run twice - Once for each value of the `logObject` configuration parameter.

calvinwiebe commented:
Code looks good. One thought I have: would it be better to _always_ log both the message and the object? So the log function would take 2 params
```javascript
timeMe.configure({log: function(msg, obj){}});
```
So you are never at a point where you want to configure two different timeMe's just to get a msg and an object. You could always massage the object into a message, of course, but it seems like a simpler/more flexible api if you always get both.

calvinwiebe commented on test/utils/index.js:
```diff
@@ -1,6 +1,46 @@
 Math.lOOms = function() {
     return parseInt(Math.random() * 100);
 }
+
```
Maybe just put a little documentation blurb here about everything that `stubLogger` does.

calvinwiebe commented:
:+1:

bllanos commented:
Updated PR description (because the code changed significantly):

### Changes

- `time-me` will pass a `{prefix, elapsed}` object as a second argument after the string version of the same information whenever it makes a call to the `log` function.

### Testing Performed
- Updated unit tests to validate new logging output.
